plugins/main/public/components/endpoints-summary/agent/index.tsx

```diff
@@ -18,11 +22,48 @@ import {
 import { useRouterSearch } from '../../common/hooks/use-router-search';
 import { Redirect, Route, Switch } from '../../router-search';
 import NavigationService from '../../../react-services/navigation-service';
+import { connect } from 'react-redux';
+import { PromptNoSelectedAgent } from '../../agents/prompts';
+import { RedirectAppLinks } from '../../../../../../src/plugins/opensearch_dashboards_react/public';
+import { getCore } from '../../../kibana-services';
+import { endpointSummary } from '../../../utils/applications';
+
+const mapStateToProps = state => ({
+  agent: state.appStateReducers?.currentAgentData,
+});
 
 export const AgentView = compose(
   withErrorBoundary,
   withRouteResolvers({ enableMenu, ip, nestedResolve, savedSearch }),
-)(() => {
+  connect(mapStateToProps),
+  withGuard(
+    props => !(props.agent && props.agent.id),
+    () => (
+      <>
+        <PromptNoSelectedAgent
+          body={
+            <>
+              You need to select an agent or return to
+              <RedirectAppLinks application={getCore().application}>
+                <EuiLink
+                  aria-label='go to Endpoint summary'
+                  href={`${endpointSummary.id}#/agents-preview`}
+                  onClick={() =>
```

Comment:
**thought**: do the usage of href and onClick give some type of problem?

Reply:
As far as I've seen, it doesn't cause a problem to have `href` and `onClick` in the `EuiLink` component. And I didn't think it was bad that the onHover link appears since the idea is that it looks like a link, even though an onClick is needed for routing.